nipyapi/security.py

```diff
@@ -917,8 +940,65 @@ def bootstrap_security_policies(service, user_identity=None,
                 auto_create=True
             )
             nipyapi.security.add_user_to_access_policy(
-                user=nifi_proxy,
+                user=nifi_proxy_user,
                 policy=pol,
                 service="registry",
                 strict=False
             )
+
+
+def create_ssl_context_controller_service(
+        parent_pg, name, keystore_file, keystore_password, truststore_file, truststore_password, 
+        key_password=None, keystore_type=None, truststore_type=None, ssl_protocol=None):
+    """
+    Creates and configures an SSL Context Service for secure client connections.
+    Note that once created it can be listed and deleted using the standard canvas functions.
+    
+    Args:
+        parent_pg (ProcessGroupEntity): The Process Group to create the service in
+        name (str): Name for the SSL Context Service
+        keystore_file (str): Path to the client certificate/keystore file
+        keystore_password (str): Password for the keystore
+        truststore_file (str): Path to the truststore file
+        truststore_password (str): Password for the truststore
+        key_password (Optional[str]): Password for the key, defaults to keystore_password if not set
+        keystore_type (Optional[str]): Type of keystore (JKS, PKCS12), defaults to JKS
+        truststore_type (Optional[str]): Type of truststore (JKS, PKCS12), defaults to JKS
+        ssl_protocol (Optional[str]): SSL protocol version, defaults to SSL
+    
+    Returns:
+        (ControllerServiceEntity): The configured SSL Context Service
+    """
+    assert isinstance(parent_pg, nipyapi.nifi.ProcessGroupEntity)
+    assert isinstance(name, six.string_types)
+    assert isinstance(keystore_file, six.string_types)
+    assert isinstance(keystore_password, six.string_types)
+    assert isinstance(truststore_file, six.string_types)
+    assert isinstance(truststore_password, six.string_types)
+    assert key_password is None or isinstance(key_password, six.string_types)
+    assert keystore_type is None or isinstance(keystore_type, six.string_types)
+    assert truststore_type is None or isinstance(truststore_type, six.string_types)
+    assert ssl_protocol is None or isinstance(ssl_protocol, six.string_types)
+
+    with nipyapi.utils.rest_exceptions():
+        return nipyapi.nifi.ControllerApi().create_controller_service(
+            body=nipyapi.nifi.ControllerServiceEntity(
+                revision=nipyapi.nifi.RevisionDTO(
+                    version=0
+                ),
+                component=nipyapi.nifi.ControllerServiceDTO(
+                    type='org.apache.nifi.ssl.StandardSSLContextService',
```

Comment:
Should this default to useing a `StandardRestrictedSSLContextService`, which is recommended over the `StandardSSLContextService` according to the [component's docs](https://github.com/apache/nifi/blob/6a2081e96524de5cbe111e2aef475b11f3145659/nifi-extension-bundles/nifi-standard-services/nifi-ssl-context-bundle/nifi-ssl-context-service/src/main/java/org/apache/nifi/ssl/StandardRestrictedSSLContextService.java#L37)

Reply:
An alternative helper method to deploy a `PEMEncodedSSLContextProvider` could also be considered, but that's probably worth a separate PR (and maybe not trying to wedge it in here as the property handling will likely get a bit confusing, e.g. the PEM files probably don't have passwords, etc.)